src/stdatamodels/jwst/datamodels/wcs_ref_models.py

```diff
@@ -343,6 +343,43 @@ def to_fits(self):
         raise NotImplementedError("FITS format is not supported for this file.")
 
 
+class MIRILrsModel(ReferenceFileModel):
+    """
+    A model for a reference file of type "specwcs" for MIRI LRS Slit.
+
+    Parameters
+    ----------
+    x_ref : float
+         x coordinate of reference position of fixed slit aperture
+    y_ref : float
+        y coordinate of reference position of fixed slit aperture
+    x_ref_slitless : float
+         x coordinate of reference position of slitless aperture
+    y_ref_slitless : float
+        y coordinate of reference position of slitless aperture
+    v2vert1 : float
+        slit vertex 1 in V2 frame
+    v2vert2 : float
+        slit vertex 2 in V2 frame
+    v2vert3 : float
+        slit vertex 3 in V2 frame
+    v2vert4 : float
+        slit vertex 4 in V2 frames
+    v3vert1 : float
+        slit vertex 1 in V3 frames
+    v3vert2 : float
+        slit vertex 2 in V3 frames
+    v3vert3 : float
+        slit vertex 3 in V3 frames
+    v3vert4 : float
+        slit vertex 4 in V3 frames
+    wavetable : numpy  2-D array
+        For each row in the slit hold  wavelength, and 
+        x center, ycenter, x and y box region cooresponding to the wavelength
+    """
+    schema_url = "http://stsci.edu/schemas/jwst_datamodel/specwcs_miri_lrs.schema"
+    reftype = "specwcs"
+       
```

Comment:
If you want to be able to programmatically create a MIRILRSModel then you need to implement the `__init__` method.

A `populate_meta` method may be useful to make sure the required meta keywords are present.

Would yo consider renaming the model to `MiriLRSSpecwcs` to be consistent with the other simialr models?

Reply:
@nden I will add __init__ and populate_meta. I will follow what the other models do in wcs_ref_models.py. I have never made a model like this. Could you explain the reason these methods are in the model. Is it to able to create 
a model in python - on the fly and not use a reference file. 

On the name I tried to name the model in a way that was consistent with the other models defined in wcs_ref_models.py.
All the models had the instrument capitalized and did not have Specwcs in the model name. Am I missing something ?

Reply:
> Is it to able to create a model in python - on the fly and not use a reference file.

Yes, that's the only reason. If you think the model will be be initialized only by a file then it's not necessary to add an `__init__`.

Reply:
@nden I have a question about how I have set up the specwcs_miri_lrs.schema.
I put all the float values that are used - x, y location of the slit frame , v2 v3 vertices in the meta data of the schema.
I am wondering if I should pull that out of meta just have entries for each float value so I would access them 
model.x_ref instead of model.meta.x_ref. 
When I call __initi___ I could pass these values and in the populate_meta - it would just be things
like instrument and exp_type.
Is that more standard ?